rio_cogeo/scripts/cli.py

```diff
@@ -153,7 +153,7 @@ def cogeo():
     "-r",
     help="Resampling algorithm (default: nearest). Will only be applied with the `--web-optimized` option.",
     type=click.Choice(
-        [it.name for it in ResamplingEnums if it.value in [0, 1, 2, 3, 4, 5, 6, 7]]
+        [it.name for it in ResamplingEnums if it.value in [0, 1, 2, 3, 4, 5, 6]]
```

Comment:
IMO this is hard to read because 0, 1, 2, etc aren't values with meaning. I know @vincentsarago already had these listed as numbers, but I would suggest with this change to list the allowed resampling enums explicitly. A simple way to do this is filter on the `name` instead of the `value`:

```py
[it.name for it in ResamplingEnums if it.name in ['Nearest', 'Bilinear', ...]
```

Reply:
that's way better @kylebarron 👍

Reply:
```suggestion
        [it.name for it in ResamplingEnums if it.name in ['nearest', 'bilinear', 'cubic', 'cubic_spline', 'lanczos', 'average', 'mode']]
```

(Note you'll have to run `black` with this probably)

Reply:
Note also that picking from `ResampingEnums` is no longer necessary... you could also use the strings directly

```suggestion
        ['nearest', 'bilinear', 'cubic', 'cubic_spline', 'lanczos', 'average', 'mode']
```

Reply:
There is another possible option: use [SUPPORTED_RESAMPLING](https://github.com/mapbox/rasterio/blob/e2add64321fdc63b39b65fc17a3a3071ead95170/rasterio/warp.py#L20) from rasterio.

Reply:
Had no idea that existed. That looks good to me, especially because it handles what resampling is available for different GDAL versions, though it's not clear it's part of rasterio's public API (i.e. it's not listed in the [docs page](https://rasterio.readthedocs.io/en/latest/api/rasterio.warp.html)). So it could conceivably break in the future if it were renamed 🤷‍♂️

Reply:
Supported Resampling was just added to Rasterio, I don't think we want to add a minimum rasterio version dependency just for this. 

FYI, I've raises the resampling issue in GDAL recently because it wasn't clear which algorithm was available for overview or for warp. In Rasterio we added https://github.com/mapbox/rasterio/blob/master/rasterio/enums.py#L98 and I feel we could also add `_WarpResampling` 🤷 

@kylebarron I'm in favour of your 1st commit suggestion. I think keeping ResamplingEnums is fine!

Reply:
`SUPPORTED_RESAMPLING` was [added](https://github.com/mapbox/rasterio/commit/1c406b4074ef9bb8a95f773a8283dce3b668ff1f) 4 years ago.

Reply:
🤦

Reply:
alright then 😂